File: app/src/main/java/com/youtility/common/base/BaseActivity.java
package com.youtility.common.base;

import android.content.Context;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

public class BaseActivity extends AppCompatActivity {

    protected Context context;
    protected String TAG = getClass().getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        context = this;
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(TAG, "started");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "stopped");
    }
}

File: energyswitchcui/src/main/kotlin/com/youtility/energyswitchcui/common/base/BaseActivity.kt
package com.youtility.energyswitchcui.common.base

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log

open class BaseActivity : AppCompatActivity() {

    protected lateinit var context: Context
    protected var TAG = javaClass.getSimpleName()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        context = this
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "started")
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "stopped")
    }
}